ludwig/datasets/dataset_config.py

```diff
@@ -29,6 +29,9 @@ class DatasetConfig:
     # The readable description of the dataset
     description: str = ""
 
+    # The learning task(s) associated with this dataset.
```

Comment:
Provide reason for this field - what is it for?
```suggestion
    # Optional.  The types of learning tasks this dataset is intended to test. Helps users discover new datasets and
    # filter for relevance to a specific machine learning setting.
```